stream.go

```diff
@@ -86,8 +86,9 @@ type ClientStream interface {
 	CloseSend() error
 	// Context returns the context for this stream.
 	//
-	// It should not be called until after Header or RecvMsg has returned. Once
-	// called, subsequent client-side retries are disabled.
+	// It should not be called until after Header or RecvMsg has returned, nor
```

Comment:
Did you see problems calling context concurrent with a Send?  I wouldn't expect any.  Did you mean to put this on CloseSend?

Reply:
> Did you see problems calling context concurrent with a Send?

I'm not sure I fully understand this question, but I did not see problems calling context's cancel function concurrently with a send, no. My problems were around calling CloseSend and SendMsg concurrently.

> Did you mean to put this on CloseSend?

No. The message applies to both CloseSend and SendMsg (they can't be used concurrently with each other), so I put it on SendMsg since there's already some discussion about concurrent calls there.

Reply:
Shooooooooot. My fault entirely, thanks for being so tactful. Removed.

Reply:
I think it makes sense to keep it, but put it on `CloseSend` instead.  I'm pretty sure that was what you meant to do and just missed it by a couple lines.

Reply:
Done